mm2src/coins/utxo/utxo_builder/utxo_arc_builder.rs

```diff
@@ -143,44 +265,14 @@ async fn merge_utxo_loop<T>(
             None => break,
         };
 
-        let my_address = match coin.as_ref().derivation_method {
-            DerivationMethod::SingleAddress(ref my_address) => my_address,
-            DerivationMethod::HDWallet(_) => {
-                warn!("'merge_utxo_loop' is currently not used for HD wallets");
-                return;
-            },
-        };
-
         let ticker = &coin.as_ref().conf.ticker;
-        let (unspents, recently_spent) = match coin.get_unspent_ordered_list(my_address).await {
-            Ok((unspents, recently_spent)) => (unspents, recently_spent),
-            Err(e) => {
-                error!("Error {} on get_unspent_ordered_list of coin {}", e, ticker);
-                continue;
-            },
-        };
-        if unspents.len() >= merge_at {
-            let unspents: Vec<_> = unspents.into_iter().take(max_merge_at_once).collect();
-            info!("Trying to merge {} UTXOs of coin {}", unspents.len(), ticker);
-            let value = unspents.iter().fold(0, |sum, unspent| sum + unspent.value);
-            let script_pubkey = Builder::build_p2pkh(my_address.hash()).to_bytes();
-            let output = TransactionOutput { value, script_pubkey };
-            let merge_tx_fut = generate_and_send_tx(
-                &coin,
-                unspents,
-                None,
-                FeePolicy::DeductFromOutput(0),
-                recently_spent,
-                vec![output],
-            );
-            match merge_tx_fut.await {
-                Ok(tx) => info!(
-                    "UTXO merge successful for coin {}, tx_hash {:?}",
-                    ticker,
-                    tx.hash().reversed()
-                ),
-                Err(e) => error!("Error {:?} on UTXO merge attempt for coin {}", e, ticker),
-            }
+        match merge_utxos(&coin, &my_address, &script_pubkey, &merge_conditions, true).await {
+            Ok((tx, spents)) => info!(
+                "UTXO merge of {} outputs successful for coin={ticker}, tx_hash={}",
+                spents.len(),
+                tx.hash().reversed()
+            ),
+            Err(e) => error!("Error on UTXO merge attempt for coin={ticker}: {e}"),
```

Comment:
Wouldn't this spam the logs with errors that are not actual when `unspents.len() < merge_conditions.merge_at` which will happen eventually.

Reply:
fixed here [dc5be54](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2587/commits/dc5be540ead85de215e09624cdb51f0ebe9dc1f5)